src/nano_contracts/utils.ts

```diff
@@ -315,3 +315,39 @@ export const getResultHelper = async <T>(
     return [e as Error, null];
   }
 };
+
+/**
+ * Retrieves the blueprint ID for a nano contract by its ID.
+ *
+ * The priority is to get the data from the getFullTxById API because
+ * it gets contract txs that are still to be confirmed by a block. If it fails,
+ * the contract might have been created by another contract, so we fallback to the state
+ * API, which gets the information correctly in that case.
+ *
+ * @returns A promise resolving to the blueprint ID.
+ * @throws {NanoContractTransactionError} If both API calls fail or if the nano contract ID is not defined.
+ * @throws {Error} If the nano contract ID is not defined.
+ */
+export const getBlueprintId = async (ncId: string, wallet: HathorWallet): Promise<string> => {
+  if (!ncId) {
+    throw new Error('Nano contract ID is not defined');
+  }
+
+  const [txError, txResponse] = (await getResultHelper(() =>
+    wallet.getFullTxById(ncId)
+  )) as [Error | null, FullNodeTxResponse | null];
+  if (!txError && txResponse?.tx?.nc_id && txResponse.tx.nc_blueprint_id) {
+    return txResponse.tx.nc_blueprint_id;
+  }
```

Comment:
Since we use the wallet only for this method and this method only calls the `txApi.getTransaction`, why not remove the wallet from the arguments and call the api directly?

Reply:
The idea is that we could use this method with the wallet service facade as well, which calls a different API for this. For now we still depend on the state API call but we should trust in the facade methods when possible.